content/client-lib-development-guide/features.textile

```diff
@@ -56,7 +56,10 @@ h2(#rest). REST client library
 
 h3(#restclient). RestClient
 
-* @(RSC1)@ The constructor accepts either an API key, a token string, or a set of "@ClientOptions@":#options.  If invalid arguments are provided such as a no API key, no token and no means to create a token, then this will result in an error
+* @(RSC1)@ The constructor accepts a set of "@ClientOptions@":#options or, in languages that support overloaded constructors, a string which may be a token string or an API key.
+** @(RSC1a)@ If a single string argument is supplied when constructing the library then the library must determine whether this is a key or a token by checking for the presence of the ':' (colon) delimiter present in an API key. Any other string must be treated as a token string.
+** @(RSC1b)@ If invalid arguments are provided such as no API key, no token and no means to create a token, then this will result in an error with error code 40000 and an informative message.
```

Comment:
@paddybyers should we not consider a new error code here?  I believe the goal with 1.1 is to provide useful error codes with URLs to specific advice on how to fix them, whereas 40000 is just a generic code.

Reply:
> Unable to obtain credentials from given parameters

Sure, that makes sense to me

Reply:
https://github.com/ably/ably-common/pull/28